### ISSUE TITLE
Undocumented members include by automodule when only :members: specified.

### ISSUE THREAD

**Issue by cjw296** [sphinx-doc/sphinx#9739]
### Describe the bug

It appears there's been a regression at some point and with v4.2.0, undocumented members are being included by `automodule`.

### How to Reproduce

Here's the source:
https://raw.githubusercontent.com/simplistix/sybil/8ba210ff3ac33a50dc3611a9153c67dae05e6919/docs/api.rst

Here's a link to the rendering:
https://sybil.readthedocs.io/en/latest/api.html#sybil.parsers.capture.DocumentReverseIterator

(although I may have to work around it...)

### Expected behavior

Undocumented members such as [DocumentReverseIterator](https://github.com/simplistix/sybil/blob/9e397645cf4bf529faa45bcd3fbcd0873a686edf/sybil/parsers/capture.py#L36-L50) and [SybilFailure](https://github.com/simplistix/sybil/blob/9e397645cf4bf529faa45bcd3fbcd0873a686edf/sybil/example.py#L10-L18) should not be included, unless `:undoc-members:` is specified.

### Your project

https://github.com/simplistix/sybil/

### OS

linux

### Python version

3.7

### Sphinx version

4.2.0

**Comment by electric-coder** [comment]
Has this been fixed? I can't find the mentioned `DocumentReverseIterator` or `SybilFailure` in the linked documentation.

**Comment by cjw296** [comment]
I don't think it has been fixed, I ended up working around this by explicitly specifying the member list:

```
capture parser
--------------

.. automodule:: sybil.parsers.capture
  :members: parse_captures
```

**Comment by malmiteria** [comment]
This bug seems to still be up.

My understanding is that classes without docstrings, but inheriting a class which itself has a docstring are included.

automodules includes classes when they inherit documented classes, even when not documented themselves.
